### PR TITLE
Templatized installer indexes

### LINKED ISSUE #3501: Automate generation of installer landing pages
Two types of landing / index pages need to be generated for most products:

1. Root "installing" landing page provides full index of all platform-specific instructions across OS and arch as well as links to subset landing pages for each OS/arch combo
2. OS/arch specific landing pages provide subset of index for topics in each section

See also: 
- https://github.com/EnterpriseDB/docs/pull/2834
- https://github.com/EnterpriseDB/docs/pull/3243

### PR DESCRIPTION
**Pull request by josh-heyer**
## What Changed?

Generate root installation index and linux CPU arch specific subindexes for templatized products.

New template patterns available:

- `products/<product>/index.njk`
- `products/<product>/v1_index.njk`
  Either of these patterns will suffice to generate an index (either for a specific version of a product or all versions)

- `products/<product>/<arch>_index.njk`
- `products/<product>/v1_<arch>_index.njk`
  Either of these patterns will suffice to generate an architecture-specific index (either for a specific version of a product or all versions)

See first commit here for examples of how these can be used.

Fixes #3501

Note: this also introduces Ubuntu 22.04 into all relevant indexes - however, individual product templates still need to be modified to add the deployment path. When deployment no longer completes with the message, `18 files skipped ` then that will also be completed.

### CODE REVIEW

File: product_docs/docs/pem/9/installing_pem_agent/index.mdx [hunk 1]
```diff
@@ -1,35 +1,49 @@
 ---
-navTitle: "Installing the PEM agent"
-title: "Installing Postgres Enterprise Manager agent"
-# This is new created file
-# Line 15 onwards need review/edits
+navTitle: Installing
```
Comment (dwicinas) [inline product_docs/docs/pem/9/installing_pem_agent/index.mdx:2]: For the navTitle, we should differentiate between installing the agent and installing the server. You can see how the old version was "Installing the PEM agent". Right now, for PEM, the nav tree just shows two "Installing" entries.

File: product_docs/docs/pgpool/4/installing_extensions/index.mdx [hunk 2]
```diff
@@ -1,33 +1,56 @@
 ---
-title: "Installing and managing Pgpool-II extensions"
-navTitle: "Installing extensions"
+navTitle: Installing
```
Comment (dwicinas) [inline product_docs/docs/pgpool/4/installing_extensions/index.mdx:2]: Same comment as I made for PEM. For the navTitle, we should differentiate between installing extensions and installing pgPool. You can see how the old version was "Installing extensions". Right now, the nav tree just shows two "Installing" entries.

File: install_template/templates/products/edb-odbc-connector/x86_64_index.njk [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+{% extends "platformBase/x86_64_index.njk" %}
+{% set productShortname="hadoop" %}
```
Comment (dwicinas) [inline install_template/templates/products/edb-odbc-connector/x86_64_index.njk:3]: short name should be "odbc"

File: install_template/templates/products/edb-odbc-connector/x86_64_index.njk [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+{% block frontmatter %}
+deployPath: odbc_connector/{{ product.version }}/installing/linux_x86_64/index.mdx
+redirects: 
+  - /hadoop_data_adapter/latest/05_installing_the_hadoop_data_adapter/x86_amd64/
```
Comment (dwicinas) [inline install_template/templates/products/edb-odbc-connector/x86_64_index.njk:8]: this looks like a cut and paste error. Needs a different path for ODBC.

File: product_docs/docs/pem/9/installing_pem_agent/index.mdx (hunk 1 above)
Comment (josh-heyer) [reply product_docs/docs/pem/9/installing_pem_agent/index.mdx]: Added a block for these - also overriding the defaults in cases where we include Windows or MacOS instructions

**Review by drothery-edb — DISMISSED** [review]

**Pull request merged**